### ISSUE TITLE
SpectaQL always returns a successful exit code, even if errors occur.

### ISSUE THREAD

**Issue by zleonov** [anvilco/spectaql#805]
Hi @newhouse and team. Let me start by saying what a great tool. I have been looking for a tool that we can hook into our build to automatically generate GraphQL documentation, and SpectaQL is the only one I found that actually works. I didn't expect it to be so polished.

Also, I wonder if you should or could enable GitHub discussions? I see some number of issues which seem more like open questions/discussions rather than specific bugs or feature requests. I don't want to pollute the Issues with open ended questions or is that ok?

Alright, not sure if you count this as a bug or feature request:

As far as I can tell SpectaQL always returns an exit code of `0`, even if it encounters errors. Currently part of our build is still handled by ant. It seems the only option is to examine the `stderr` property of the `exec` task to determine whether or not to fail the build. This seems less than preferable and error prone. I am not sure what may end up in the `stderr` stream, maybe warning, that should simply be ignored?

Am I missing something? Should I expect SpectaQL to return an error code, for example if it fails while parsing the schema?

**Comment by newhouse** [comment]
@zleonov thanks for the kind words, and thank you for using SpectaQL!

Can you provide some example output of an error you're seeing, but where the return code is `0`?

The behavior you're expecting sounds reasonable to me, so I will probably try to do this at some point in the near future

**Comment by zleonov** [comment]
Sure, so for example any schema error. I discovered via SpectaQL that I had a duplicate value in an enum type. Apparently the framework we are using is more forgiving, it didn't throw the error, but SpectaQL did.

I just created a syntax error and ran the doc target in ant, here is the output (ignore the `[echo]` part, that's from ant), as you can see the build is successful even though there is an error, when I manually check why, it's because the return code is `0` indicating success:

```
generate-graphql-schema-docs:
     [echo] Syntax Error: Expected Name, found ":".
     [echo] 
     [echo] GraphQL request:41:11
     [echo] 40 |         """
     [echo] 41 |     hello:: String!
     [echo]    |           ^
     [echo] 42 |         """
     [echo]     at syntaxError (C:\...\spectaql\node_modules\graphql\error\syntaxError.js:15:10)
     [echo]     at Parser.expectToken (C:\...\spectaql\node_modules\graphql\language\parser.js:1397:40)
     ...
     [echo]     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
     [echo]     at node:internal/main/run_main_module:23:47
     [echo]  {
     [echo]   path: undefined,
     [echo]   locations: [ { line: 41, column: 11 } ],
     [echo]   extensions: [Object: null prototype] {}
     [echo] }

BUILD SUCCESSFUL
Total time: 5 seconds
```

I suppose schema parsing errors probably cover 90% of the possible errors that can occur. But also I assume the yaml config file can cause error if it has syntax mistakes.

**Comment by zleonov** [comment]
I mean it could be any error. For example if the schema file is not found. Or if the target directory does not have writable permissions for the account running the build.

I am not sure if any non-fatal warnings are written to the stderr. If not, one strategy would be simple: any place where anything is written to stderr should set an error exit code.

**Comment by newhouse** [comment]
Yah, no problem...should be easy to repo.

**Comment by zleonov** [comment]
Update. As I was playing around with customizing the css styles I discovered that SpectaQL does, in some cases return an error code. However, the error itself was printed to stdout not stderr. Maybe this is a result of a submodule? To replicate, simply create a syntax error in custom.scss.

Here is the output:
```
generate-graphql-schema-docs:
     [exec] Result: 1
     ....
     [echo] Running "sass:main" (sass) task
     [echo] Fatal error: Error: expected "}".
     [echo]    ╷
     [echo] 34 │   }
     [echo]    │    ^
     [echo]    ╵
     [echo]   C:\Users\eleonov\AppData\Local\Temp\spectaqltmp--20140-AecSXLIWwm4q\stylesheets\custom.scss 34:4  @import
     [echo]   C:\Users\eleonov\AppData\Local\Temp\spectaqltmp--20140-AecSXLIWwm4q\stylesheets\main.scss 69:9    root stylesheet
```
Again, although you can't see from the snippet above, it's interesting that this error was in stdout not stderr.

The overall issue remains the same. Perhaps we should name it: `Consistent error return code and output`. In the meantime I think my condition for failing the build should be: either a non-zero exit code or stderr in empty.

**Comment by newhouse** [comment]
This is fixed in `main` and will be out shortly in `2.2.0`. Please confirm @zleonov
